nixos/modules/system/boot/initrd-ssh.nix

```diff
@@ -164,13 +164,12 @@ in
           for instructions.
         '';
       }
-
-      {
-        assertion = config.boot.initrd.systemd.enable -> cfg.shell == null;
-        message = "systemd stage 1 does not support boot.initrd.network.ssh.shell";
-      }
     ];
 
+    warnings = lib.optional (config.boot.initrd.systemd.enable -> cfg.shell != null) ''
```

Comment:
I don't think this is correct.

```
nix-repl> nixosConfigurations.hydrogen.config.boot.initrd.systemd.enable -> nixosConfigurations.hydrogen.config.boot.initrd.network.ssh.shell != null
true

nix-repl> nixosConfigurations.hydrogen.config.boot.initrd.network.ssh.shell
null

nix-repl> nixosConfigurations.hydrogen.config.boot.initrd.systemd.enable
false
```

Reply:
I think you're right. When transitioning from assertion to warning, I switched the `==` with `!=`, which is good, but I didn't change `->` to `&&`.

Reply:
@SuperSandro2000 #266116